### PR TITLE
Add table aws_codebuild_build

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
# Integration test logs
<details>
  <summary>Logs</summary>

```
No integration test exist.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
select
  arn,
  id,
  build_complete,
  timeout_in_minutes,
  project_name,
  build_status,
  encryption_key,
  end_time,
  region
from
  aws_codebuild_build;
+--------------------------------------------------------------------------------------------------+---------------------------------------------------+----------------+--------------------+--------------+--------------+--------------------------------------------------+>
| arn                                                                                              | id                                                | build_complete | timeout_in_minutes | project_name | build_status | encryption_key                                   |>
+--------------------------------------------------------------------------------------------------+---------------------------------------------------+----------------+--------------------+--------------+--------------+--------------------------------------------------+>
| arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9    | test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9     | true           | 60                 | test-bp1     | "FAILED"     | arn:aws:kms:ap-south-1:222222222222:alias/aws/s3 |>
| arn:aws:codebuild:ap-south-1:222222222222:build/test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9      | test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9       | true           | 60                 | test-p       | "FAILED"     | arn:aws:kms:ap-south-1:222222222222:alias/aws/s3 |>
| arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40    | test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40     | true           | 60                 | test-bp1     | "FAILED"     | arn:aws:kms:ap-south-1:222222222222:alias/aws/s3 |>
| arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0    | test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0     | true           | 60                 | test-bp1     | "FAILED"     | arn:aws:kms:ap-south-1:222222222222:alias/aws/s3 |>
| arn:aws:codebuild:ap-south-1:222222222222:build/test-2:8862f58c-edeb-4789-a302-d56057204ada      | test-2:8862f58c-edeb-4789-a302-d56057204ada       | true           | 60                 | test-2       | "FAILED"     | arn:aws:kms:ap-south-1:222222222222:alias/aws/s3 |>
| arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053    | test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053     | true           | 60                 | test-bp1     | "FAILED"     | arn:aws:kms:ap-south-1:222222222222:alias/aws/s3 |>
| arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd    | test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd     | true           | 60                 | test-bp1     | "SUCCEEDED"  | arn:aws:kms:ap-south-1:222222222222:alias/aws/s3 |>
| arn:aws:codebuild:us-east-2:222222222222:build/testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | true           | 60                 | testbp12     | "FAILED"     | arn:aws:kms:us-east-2:222222222222:alias/aws/s3  |>
| arn:aws:codebuild:us-east-2:222222222222:build/test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | true           | 5                  | test-cb12345 | "FAILED"     | arn:aws:kms:us-east-2:222222222222:alias/aws/s3  |>
+--------------------------------------------------------------------------------------------------+---------------------------------------------------+----------------+--------------------+--------------+--------------+--------------------------------------------------+>

select
  id,
  arn,
  artifacts,
  build_complete
from
  aws_codebuild_build
where
  build_complete;
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------->
| id                                                | arn                                                                                              | artifacts                                                                                                             >
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------->
| test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9    | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>
| test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9       | arn:aws:codebuild:ap-south-1:222222222222:build/test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9      | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":false,"Location":"arn:aws:s3:::aa-parth-test/te>
| test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40    | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>
| test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0    | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>
| test-2:8862f58c-edeb-4789-a302-d56057204ada       | arn:aws:codebuild:ap-south-1:222222222222:build/test-2:8862f58c-edeb-4789-a302-d56057204ada      | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>
| test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053    | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>
| test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd    | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>
| testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | arn:aws:codebuild:us-east-2:222222222222:build/testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>
| test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | arn:aws:codebuild:us-east-2:222222222222:build/test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | {"ArtifactIdentifier":null,"BucketOwnerAccess":"","EncryptionDisabled":null,"Location":"","Md5sum":null,"OverrideArtif>

select
  id,
  arn,
  vpc_config ->> 'SecurityGroupIds' as security_groups,
  vpc_config ->> 'Subnets' as subnets,
  vpc_config ->> 'VpcId' as vpc_id
from
  aws_codebuild_build;
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+--------------------------+----------------------------------------------------------+-----------------------+
| id                                                | arn                                                                                              | security_groups          | subnets                                                  | vpc_id                |
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+--------------------------+----------------------------------------------------------+-----------------------+
| test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9    | <null>                   | <null>                                                   | <null>                |
| test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9       | arn:aws:codebuild:ap-south-1:222222222222:build/test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9      | <null>                   | <null>                                                   | <null>                |
| test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40    | <null>                   | <null>                                                   | <null>                |
| test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0    | <null>                   | <null>                                                   | <null>                |
| test-2:8862f58c-edeb-4789-a302-d56057204ada       | arn:aws:codebuild:ap-south-1:222222222222:build/test-2:8862f58c-edeb-4789-a302-d56057204ada      | <null>                   | <null>                                                   | <null>                |
| test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053    | <null>                   | <null>                                                   | <null>                |
| test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd    | <null>                   | <null>                                                   | <null>                |
| testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | arn:aws:codebuild:us-east-2:222222222222:build/testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | <null>                   | <null>                                                   | <null>                |
| test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | arn:aws:codebuild:us-east-2:222222222222:build/test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | ["sg-0024adac213a69357"] | ["subnet-05ec8288f0b9be5aa", "subnet-0b349fd9ce6590352"] | vpc-0a93262e0a9f10dda |
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+--------------------------+----------------------------------------------------------+-----------------------+

select
  id,
  arn,
  artifacts ->> 'ArtifactIdentifier' as artifact_id,
  artifacts ->> 'BucketOwnerAccess' as bucket_owner_access,
  artifacts ->> 'EncryptionDisabled' as encryption_disabled,
  artifacts ->> 'OverrideArtifactName' as override_artifact_name
from
  aws_codebuild_build;
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+-------------+---------------------+---------------------+------------------------+
| id                                                | arn                                                                                              | artifact_id | bucket_owner_access | encryption_disabled | override_artifact_name |
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+-------------+---------------------+---------------------+------------------------+
| test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:01ad56f8-0790-4b9d-a354-dec00c310d40    | <null>      |                     | <null>              | <null>                 |
| test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9       | arn:aws:codebuild:ap-south-1:222222222222:build/test-p:ddbc61bd-2f5e-4607-808d-54bcb47462f9      | <null>      |                     | false               | false                  |
| test-2:8862f58c-edeb-4789-a302-d56057204ada       | arn:aws:codebuild:ap-south-1:222222222222:build/test-2:8862f58c-edeb-4789-a302-d56057204ada      | <null>      |                     | <null>              | <null>                 |
| test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:5befbc41-36ae-45d1-9d3a-f12525622fa9    | <null>      |                     | <null>              | <null>                 |
| test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4d332dab-2d9c-4b04-8cd9-2a75bb5de053    | <null>      |                     | <null>              | <null>                 |
| test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:1c3e858b-48fd-40f4-954f-1e86f237b3dd    | <null>      |                     | <null>              | <null>                 |
| test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0     | arn:aws:codebuild:ap-south-1:222222222222:build/test-bp1:4a836a33-9f56-4247-a06e-b62e60b7b8d0    | <null>      |                     | <null>              | <null>                 |
| test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | arn:aws:codebuild:us-east-2:222222222222:build/test-cb12345:16d11b0f-c816-4885-b292-b21197f53568 | <null>      |                     | <null>              | <null>                 |
| testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | arn:aws:codebuild:us-east-2:222222222222:build/testbp12:fcaeba73-401c-4d3c-8805-22b0552e1caf     | <null>      |                     | <null>              | <null>                 |
+---------------------------------------------------+--------------------------------------------------------------------------------------------------+-------------+---------------------+---------------------+------------------------+
```
</details>

### CODE REVIEW

**Comment by Priyanka-Chatterjee-2000** [comment]
This PR is in draft state as on querying `select * from aws_codebuild_build`
It is giving old build data but I am not getting the build details of currently created codebuild project. Is it a code error or the data is fetched that way by the api itself?

File: aws/table_aws_codebuild_build.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+// to get any other info an API call per codebuild build would need to be made. So in the case where we need to get
+// all info for less then 100 instances including the BatchGetBuild request here, and batching requests means only making
+// two API calls as opposed to 101.
+func listCodeBuildBuilds(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listCodeBuildBuilds")
+
+	// Create Session
+	svc, err := CodeBuildClient(ctx, d)
+	if err != nil {
+		return nil, err
```
Comment (ParthaI) [inline aws/table_aws_codebuild_build.go:210]: Please add error log

**Review by ParthaI — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please see the comment.
If we could add more example query it would be more helpful

Thanks!

**Review by ParthaI — APPROVED** [review]
LGTM

File: aws/table_aws_codebuild_build.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+			{
+				Name:        "source",
+				Description: "Information about the build input source code for the build project.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "vpc_config",
+				Description: "Information about the VPC configuration that CodeBuild accesses.",
+				Type:        proto.ColumnType_JSON,
+			},
```
Comment (misraved) [inline aws/table_aws_codebuild_build.go:178]:
```suggestion
			},
			
			// Steampipe standard columns
```

File: aws/table_aws_codebuild_build.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Arn"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(transform.EnsureStringArray),
+			},
```
Comment (misraved) [inline aws/table_aws_codebuild_build.go:190]: Does the resource support tags?

File: aws/table_aws_codebuild_build.go [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+// to get any other info an API call per codebuild build would need to be made. So in the case where we need to get
+// all info for less then 100 instances including the BatchGetBuild request here, and batching requests means only making
+// two API calls as opposed to 101.
+func listCodeBuildBuilds(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listCodeBuildBuilds")
+
+	// Create Session
+	svc, err := CodeBuildClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_codebuild_build.listCodeBuildBuild", "get_client_error", err)
```
Comment (misraved) [inline aws/table_aws_codebuild_build.go:210]:
```suggestion
		plugin.Logger(ctx).Error("aws_codebuild_build.listCodeBuildBuilds", "connection_error", err)
```

File: aws/table_aws_codebuild_build.go [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+	paginator := codebuild.NewListBuildsPaginator(svc, input, func(o *codebuild.ListBuildsPaginatorOptions) {
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_codebuild_build.listCodeBuildBuild", "api_error", err)
```
Comment (misraved) [inline aws/table_aws_codebuild_build.go:224]:
```suggestion
			plugin.Logger(ctx).Error("aws_codebuild_build.listCodeBuildBuilds", "api_error", err)
```

File: docs/tables/aws_codebuild_build.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  arn,
+  id,
+  encryption_key
+from
+  aws_codebuild_build
+where
+  encryption_key is not null;
+```
+
+### List complete build
```
Comment (misraved) [inline docs/tables/aws_codebuild_build.md:37]:
```suggestion
### List complete builds
```

File: docs/tables/aws_codebuild_build.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  arn,
+  artifacts,
+  build_complete
+from
+  aws_codebuild_build
+where
+  build_complete;
+```
+
+### List VPC configuration details of build
```
Comment (misraved) [inline docs/tables/aws_codebuild_build.md:51]:
```suggestion
### List VPC configuration details of builds
```

File: docs/tables/aws_codebuild_build.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  build_complete;
+```
+
+### List VPC configuration details of build 
+
+```sql
+select
+  id,
+  arn,
+  vpc_config ->> 'SecurityGroupIds' as security_groups,
```
Comment (misraved) [inline docs/tables/aws_codebuild_build.md:57]:
```suggestion
  vpc_config ->> 'SecurityGroupIds' as security_group_id,
```

File: docs/tables/aws_codebuild_build.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  vpc_config ->> 'Subnets' as subnets,
+  vpc_config ->> 'VpcId' as vpc_id
+from
+  aws_codebuild_build;
+```
+
+### List artifacts details of build
```
Comment (misraved) [inline docs/tables/aws_codebuild_build.md:64]:
```suggestion
### List artifact details of builds
```

File: aws/table_aws_codebuild_build.go [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+
+// listCodeBuildBuilds handles both listing and describing of the codebuild builds.
+//
+// The reason for this is the BatchGetBuilds call can accept up to 100 IDs. If we moved it out to another
+// hydrate functions we may save a request or two if we only wanted to retrieve the IDs but the tradeoff is we need
+// to get any other info an API call per codebuild build would need to be made. So in the case where we need to get
+// all info for less then 100 instances including the BatchGetBuild request here, and batching requests means only making
+// two API calls as opposed to 101.
+func listCodeBuildBuilds(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listCodeBuildBuilds")
```
Comment (misraved) [inline aws/table_aws_codebuild_build.go:205]: Is this needed?

**Review by misraved — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please take a look at the review comments. Thanks!!!

Also, I believe we have quite a few interesting columns that could be used for the example queries 👍.

File: aws/table_aws_codebuild_build.go (hunk 3 above)
Comment (Priyanka-Chatterjee-2000) [reply aws/table_aws_codebuild_build.go:190]: No

File: aws/table_aws_codebuild_build.go [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+			},
+			{
+				Name:        "project_name",
+				Description: "The name of the build project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "queued_timeout_in_minutes",
+				Description: "Specifies the amount of time, in minutes, that a build is allowed to be queued before it times out.",
+				Type:        proto.ColumnType_INT,
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply aws/table_aws_codebuild_build.go:72]: Yes it is of type *int32

**Comment by misraved** [comment]
@Priyanka-Chatterjee-2000 @ParthaI Can we add a `GetConfig` to the table wherein we are using the `id` column as a required qual?

**Comment by misraved** [comment]
@ParthaI Can we not use the `BatchGetBuilds` function with a single ID as input? Reference - https://docs.aws.amazon.com/codebuild/latest/APIReference/API_BatchGetBuilds.html

**Comment by misraved** [comment]
Alternatively @ParthaI, we could add `id` as an optional column in the `ListConfig` which would make the table code look more compact.

**Review by misraved — APPROVED** [review]

**Pull request merged**